feat: add support for multiple nacos instances and replace events library with shdict (#12263)
apache/apisix: docs/en/latest/discovery/nacos.md

### Configuration for Nacos

> There are two ways to configure Nacos. Single instance and Multi instance method.

Reviewer: ```suggestion
Nacos can be configured in two ways: single instance or multi-instance.
```

> Add following configuration in `conf/config.yaml` :

Reviewer: Can remove this sentence and just add "title", e.g. 

```
```yaml title="conf/config.yaml"
```

to the single and multi instances examples

> #### Multi Instance method

Reviewer: Put the single instance before multi, so the reading flows better because 

<img width="676" alt="image" src="https://github.com/user-attachments/assets/abb9d6a4-5b4c-459b-a9a2-0bc3bafaef7d" />

#### L7

> Here is an example of routing a request with an URI of "/nacos/*" to a service which named "http://192.168.33.1:8848/nacos/v1/ns/instance/list?serviceName=APISIX-NACOS" configured with id `nacos-cluster-1` and use nacos discovery client in the registry:

Reviewer: ```suggestion
The following is an example of routing requests matching the URI `/nacos/*` to a service at `http://192.168.33.1:8848/nacos/v1/ns/instance/list?serviceName=APISIX-NACOS`, configured with the ID `nacos-cluster-1` and using the Nacos discovery client in the registry.”
```

:::note

> You can fetch the `admin_key` from `config.yaml` and save to an environment variable with the following command:

Reviewer: Not the right place to have this note under L7. Move it one level up.

Author: Do you mean the :::note section? It was here already. Do you want me to move it?
